vkd3d: 1.10 -> 1.12

AndersonTorres opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Closes https://github.com/NixOS/nixpkgs/pull/293902

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

AndersonTorres committed:
vkd3d: migrate to by-name

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3605

SuperSamus commented:
Small note: vkd3d-proton 2.12 is likely to release soon, given that the changelog was already committed.

AndersonTorres commented:
I will stack it when it arrives, then.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3609

SuperSandro2000 commented on pkgs/by-name/vk/vkd3d-proton/package.nix:
```diff
@@ -0,0 +1,55 @@
+
+  strictDeps = true;
+
+  prePatch = let
```
```suggestion
  postPatch = let
```
We don't want to break patches

SuperSandro2000 commented on pkgs/by-name/vk/vkd3d-proton/package.nix:
```diff
@@ -0,0 +1,55 @@
+    owner = "HansKristian-Work";
+    repo = "vkd3d-proton";
+    rev = "ffb04fb7091f932ca95da236d63ab9a4064d6443";
+    fetchSubmodules = true;
```
Is there a point in devendoring things?

AndersonTorres replied:
Well, it can possibly work. I will try.

AndersonTorres replied:
Splitted at https://github.com/NixOS/nixpkgs/pull/319877

AndersonTorres committed:
vkd3d: refactor

- finalAttrs
- split outputs
- strictDeps
- no nested with
- meta
  - longDescription
  - mainProgram
  - add AndersonTorres as maintainer

AndersonTorres committed:
vkd3d: 1.10 -> 1.11

Co-authored-by: R. RyanTM <ryantm-bot@ryantm.com>

AndersonTorres committed:
vkd3d: 1.11 -> 1.12

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4091

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4095

SuperSandro2000 commented on pkgs/by-name/vk/vkd3d/package.nix:
```diff
@@ -0,0 +1,63 @@
+    bison
+    flex
+    pkg-config
+    wine
```
wine as a nativeBuildInput? I am a bit surprised but with strictDeps it is probably fine.

AndersonTorres replied:
A typical SDL situation...

SuperSandro2000 replied:
I am hearing for the first time that this is SDL related. How does it come?

AndersonTorres replied:
I said in the sense SDL requires both nativeBuildInputs and buildInputs, because each has a distinct role.

SuperSandro2000 reviewed (approved):
LGTM
